lib/components/user/places/place.js

```diff
@@ -0,0 +1,129 @@
+import { Button } from 'react-bootstrap'
+import { useIntl } from 'react-intl'
+import PropTypes from 'prop-types'
+import React from 'react'
+import styled from 'styled-components'
+
+import { LinkContainerWithQuery } from '../../form/connected-links'
+import BaseIcon from '../../util/icon'
+
+const Container = styled.li`
+  align-items: stretch;
+  display: flex;
+`
+export const PlaceButton = styled(Button)`
+  background: none;
+  flex: 1 0 0;
+  overflow: hidden;
+  text-align: left;
+  text-overflow: ellipsis;
+`
+
+export const PlaceDetail = styled.span``
+
+export const PlaceContent = styled.span``
+
+export const PlaceName = styled.span``
+
+export const PlaceText = styled.span``
+
+export const Icon = styled(BaseIcon)``
+
+export const ActionButton = styled(Button)`
+  background: none;
+  height: 100%;
+`
+
+export const ActionButtonPlaceholder = styled.span``
+
+/**
+ * Renders a stylable clickable button for editing/selecting a user's favorite place,
+ * and buttons for viewing and deleting the place if corresponding handlers are provided.
+ */
+const Place = ({
+  className,
+  detailText,
+  icon,
+  largeIcon,
+  mainText,
+  onClick,
+  onDelete,
+  onView,
+  path,
+  title = `${mainText}${detailText && ` (${detailText})`}`
+}) => {
+  const intl = useIntl()
+  const viewStopLabel = intl.formatMessage({ id: 'components.Place.viewStop' })
+  const deletePlaceLabel = intl.formatMessage({
+    id: 'components.Place.deleteThisPlace'
+  })
+  const to = onClick ? null : path
+  const iconSize = largeIcon && '2x'
+  return (
+    <Container className={className}>
+      <LinkContainerWithQuery
+        // Don't highlight component if 'to' is null.
+        activeClassName=""
+        to={to}
+      >
+        <PlaceButton onClick={onClick} title={title}>
+          {largeIcon && <Icon size="2x" type={icon} />}
```

Comment:
Could we be using `iconSize` here?

Reply:
Changed in https://github.com/opentripplanner/otp-react-redux/pull/329/commits/5f9795a0b3f3911ee4173702d7a4c11a97bcb022.